### PR TITLE
fix: Don't over-reload nginx for suspends and archives

### PR DESCRIPTION
**Pull request by balamurali27**
image

As tested on 4 core (for more nginx processes) machine and running 1000+ jobs

### CODE REVIEW

File: agent/proxy.py [hunk 1]
```diff
@@ -233,14 +237,26 @@ def remove_redirect(self, host):
...
+            )
+        )
+        if shutting_down_processes > 0:
+            return True
+
     @step("Reload NGINX")
-    def reload_nginx(self):
+    def reload_nginx(self, skip_if_reloading=False):
+        if skip_if_reloading and self.is_nginx_reloading():
+            return
```
Comment (ankush) [inline agent/proxy.py:252]: If you don't reload here, you still somehow need to reload one more time at the end of everything right?

Alternate:
- for bulk config regen - keep regenerating configs many times
- only issue one reload at end when done regenerating. Doesn't matter MUCH for suspend/archive anyway afaik.

@adityahase

Comment (balamurali27) [reply agent/proxy.py:252]: > Doesn't matter MUCH for suspend/archive anyway

Idea was this only for skipping the reload this way. Worst case some suspended/archived will get 404 till the next reload

File: agent/proxy.py [hunk 2]
```diff
@@ -233,14 +237,26 @@ def remove_redirect(self, host):
...
         return self.execute("sudo systemctl reload nginx")
 
     @step("Generate NGINX Configuration")
     def generate_proxy_config(self):
         return self._generate_proxy_config()
 
+    @cached(cache=TTLCache(maxsize=128, ttl=10))
     def _generate_proxy_config(self):
```
Comment (ankush) [inline agent/proxy.py:260]: Should this really be cached? Rendering template doesn't cost much and you might end up not re-generating when it's actually required.

Comment (balamurali27) [reply agent/proxy.py:260]: ah yeah, good catch. Left this in after testing. Not needed.

**Pull request merged**